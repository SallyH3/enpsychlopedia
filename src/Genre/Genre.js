import React, { Component } from 'react';
import Bands from '../Bands/Bands.js';
import Controls from '../Controls/Controls.js';

export default class Genre extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentBands: 'all'
    }
  }
 
  getActivity = activity => {
    this.setState({currentBands: activity})
  }

  render() {
    let bandsOrganized = this.props.sortedBands;
    if (!bandsOrganized) { 
      return (
        <h1>Hello world</h1>
      )
    }

   return (
    <section className='genre-container'>
      <div className='genre-title-container'>
        <div className='genre-title'>
          <h1>{this.props.genre.genre.toLowerCase()}</h1>
          <div className='characteristics-container'>Characteristics: &nbsp; 
            {this.props.genre.characteristics.map((characteristic, i) => {
              if(i === this.props.genre.characteristics.length - 1) {
              return <h3 className='characteristic'>{characteristic}</h3>
              } else {
              return <h3 className='characteristic'>{characteristic}, &nbsp;</h3>
              }
            }
          )}
          </div>
        </div>
        <div>
        <Controls getActivity={this.getActivity}
        />
        </div>
      </div>
      <h1>{this.props.genreIndex}</h1>
      <Bands 
        genreBands={this.props.sortedBands}
        currentSelection={this.state.currentBands}
      />
    </section>
  )
  }
}