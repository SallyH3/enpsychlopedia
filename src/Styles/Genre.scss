.genre-container {
  border: 1px solid black;
  margin: 40px;
}

.genre-title-container {
  display: flex;
  justify-content: space-between;
}