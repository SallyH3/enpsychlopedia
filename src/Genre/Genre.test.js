import React from 'react';
import Genre from './Genre';
import { shallow } from 'enzyme';

describe('Genre', () => {
  let wrapper
  beforeEach(()=> {
    wrapper = shallow(
    <Genre/>
    )
  });

  it('should match snapShot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});