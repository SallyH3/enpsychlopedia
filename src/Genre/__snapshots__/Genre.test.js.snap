// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre should match snapShot 1`] = `
<h1>
  Hi Sally!
</h1>
`;
